lib/facter/facts/solaris/os/name.rb

```diff
@@ -9,7 +9,15 @@ class Name
 
         def call_the_resolver
           value = Facter::Resolvers::Uname.resolve(:kernelname)
-          fact_value = value == 'SunOS' ? 'Solaris' : value
+          version = Facter::Resolvers::Uname.resolve(:kernelversion)
```

Comment:
This line is duplicate
```suggestion
```

Reply:
No it's not a duplicate - :kernelversion and :kernelname is not the same